src/devices/SocketCan/SocketCan.csproj

```diff
@@ -1,7 +1,7 @@
 ﻿<Project Sdk="Microsoft.NET.Sdk">
 
   <PropertyGroup>
-    <TargetFrameworks>net5.0;netcoreapp2.1</TargetFrameworks>
+    <TargetFrameworks>net5.0;netcoreapp3.1</TargetFrameworks>
```

Comment:
Projects need to maintain these two targets: 5.0 or 2.1. 

That's what IoT.Device.Bindings targets: https://github.com/dotnet/iot/blob/master/src/Iot.Device.Bindings/Iot.Device.Bindings.csproj

Certainly, a project could add a third target, but it seems like a very niche need. In the general case, bindings should provide an implementation within the oldest targeted version and feel free to light up for the newest one.

Fair?

Reply:
can we dump 2.1 support for vNext and start with 3.1? I'd prefer we didn't get stuck in the past. Perhaps we should consider splitting the Iot.Device.Bindings into individual packages so this problem is scoped

Reply:
Given: .NET Core 2.1 goes EOL in August 2021 ([source](https://github.com/dotnet/core/blob/master/microsoft-support.md)].

I have a new idea for how to approach this.

- IoT.Device.Binding should target 2.1, 3.1 and 5.0, starting now.
- New bindings are free to pick their lowest supported version (with some justification as to why).
- In August, we drop the 2.1 build / package asset as a kind of non-event.

The benefit of this plan is that the 2.1 -> 3.1 transition (as the lowest supported version) is a long process, not a single moment. It enables us to take advantage of new functionality sooner, while offering 2.1 support until the end of that release's support period.

Reply:
I would like to discuss that a bit further before we make any decision here. Even if 2.1 goes EOL, we are a library which is why in principle we want to continue targeting the lowest possible framework whenever possible. I also don't think that bringing in a build for 3.1 would be beneficial unless the 3.1 implementation is different than the 5.0 one. Basic principles should be:
- Keep targeting the lowest possible framework you can, which today is 2.1
- Only add new TFM targets on Iot.Device.Bindings if we have a) bindings that can only be supported on higher TFMs than the ones we compile against today, or b) bindings that will have a drasticly different implementation on a higher TFM given to new types that are available.

Given that, and in order to take this PR, I would probably just add a new 3.1 target on Iot.Device.Bindings for now, and condition all of the code added in this PR to be 3.1 specific, and keep the 2.1 code as it was before.

Reply:
@joperezr people really needing 2.1 can use older version of the package, the old code continues to work. If you write new code you should use newer stack and newer version of the package. I'd rather announce up to which version we support 2.1

Reply:
We'll discuss this further in our next triage meeting. There are no current downsides of multi-targeting, and there are MANY downsides for dropping support.